tests/Core/Content/Product/SalesChannel/Listing/ProductSearchRouteTest.php

```diff
@@ -41,4 +59,91 @@ protected function getOriginal()
     {
         return $this->original;
     }
+
+    public function queryProvider(): array
+    {
+        return [
+            'Searching for first variant number' => [
+                'query' => 'FINDOLOGIC001.1',
+                'expectedProductNumber' => 'FINDOLOGIC001.1'
+            ],
+            'Searching for second variant number' => [
+                'query' => 'FINDOLOGIC001.2',
+                'expectedProductNumber' => 'FINDOLOGIC001.2'
+            ],
+            'Searching for product title' => [
+                'query' => 'FINDOLOGIC Product',
+                'expectedProductNumber' => 'FINDOLOGIC001'
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider queryProvider
+     */
+    public function testSearchingVariantProductNumberWillShowVariant(string $query, string $expectedProductNumber): void
+    {
+        $this->salesChannelContext = $this->getMockedSalesChannelContext(true);
+        $product = $this->createTestProduct([], true);
+        $variantId = md5($query);
```

Comment:
Please avoid using `md5` even if it is convenient. You can also use any ambiguous string that you like.